default_validator/src/lib.rs

```diff
@@ -171,26 +197,30 @@ pub fn validate_transaction(
     authority: AccountId,
     transaction: VersionedSignedTransaction,
     block_height: u64,
-) -> Result {
+) -> Result<TransactionValidationOutput> {
```

Comment:
IMO validation functions of the executor should never return output value. Output value should be cached on the host side when the host computes it (which happens on `.execute`). The return value should not go through executor, this is just a waste of resoureces

Reply:
Our Executor already validates transaction by executing all instructions. And after that host system will execute transaction for a second time.

I think it makes sence for Executor to actually return the result of execution, if it already have to execute transaction.

Reply:
> And after that host system will execute transaction for a second time.

No, we removed second execution

Reply:
and return it to who, it's just pipelined from host to Executor and back to host